app/experimenter/experiments/views.py

```diff
@@ -525,14 +525,14 @@ def form_valid(self, form):
         return response
 
     def form_invalid(self, form):
-        return redirect(
-            "{url}?normandy_id={value}".format(
-                url=reverse(
-                    "experiments-detail", kwargs={"slug": self.kwargs["slug"]}
-                ),
-                value=form.data["normandy_id"],
-            )
+        url = reverse(
+            "experiments-detail", kwargs={"slug": self.kwargs["slug"]}
         )
+        query_parameters = form.data.copy()
```

Comment:
Why do you have to copy? You can't pop from the original?

Reply:
@jaredkerim  the original is immutable...